src/billing/components/marketplace/MarketplaceLink.tsx

```diff
@@ -0,0 +1,70 @@
+import React, {FC} from 'react'
+import {
+  Button,
+  ComponentColor,
+  ComponentSize,
+  ButtonShape,
+} from '@influxdata/clockface'
+
+import {Marketplace} from 'src/types/marketplace'
+
+// Utils
+import {useBilling} from 'src/billing/components/BillingPage'
+
+const buttonInfo = (marketplace: Marketplace): {link: string; text: string} => {
+  switch (marketplace) {
+    case 'aws':
+      return {
+        link: 'https://aws.amazon.com/marketplace/library/',
+        text: 'Check My AWS Subscription',
+      }
+    case 'gcm':
+      return {
+        link: 'https://console.cloud.google.com/marketplace/yourSolutions',
+        text: 'Check My Google Cloud Marketplace Subscription',
+      }
+    case 'azure':
+      return {
+        link:
+          'https://portal.azure.com/#blade/Microsoft_Azure_Billing/SubscriptionsBlade',
+        text: 'Check My Azure Marketplace Subscription',
+      }
+    default:
+      return {link: '', text: ''}
+  }
+}
+
+const marketplaceLinkText = marketplace => {
+  switch (marketplace) {
+    case 'aws':
+      return 'AWS Subscription Portal'
+    case 'gcp':
+      return 'Google Cloud Solution'
+    case 'azure':
+      return 'Microsoft Azure Portal'
+  }
+}
```

Comment:
#optional Again, I know this is straight from Quartz, but would it be worth combining these two functions into a single case statement with a `linkText` key in the resulting object?

Reply:
Sounds good!